feat(linalg): replace nalgebra SVD in umeyama with internal svd3

AnilKumarSingh9856 opened the pull request:
Closes #452

This PR follows up on the conversation in issue #452 and the recent merge of the new `svd3` function.

The `umeyama` function in `kornia-linalg/src/rigid.rs` has been updated to remove the `nalgebra` dependency. It now uses the internal `kornia_linalg::svd3` function for the SVD calculation.

All tests for `kornia-linalg` pass, confirming the implementation is correct.

edgarriba commented on crates/kornia-linalg/src/rigid.rs:
```diff
@@ -33,77 +26,59 @@ pub fn umeyama(src: &[Vec3], dst: &[Vec3]) -> UmeyamaResult {
-    // Centroids
-    let mut mu_s = [0.0; 3];
-    let mut mu_d = [0.0; 3];
+    // --- Centroids (using Vec3 as requested) ---
```
no agentic comments, follow rust good practices

edgarriba commented on crates/kornia-linalg/src/rigid.rs:
```diff
@@ -33,77 +26,59 @@ pub fn umeyama(src: &[Vec3], dst: &[Vec3]) -> UmeyamaResult {
+        let sc = src[i] - mu_s;
+        let dc = dst[i] - mu_d;
+        // H = sum(dc * sc.transpose())
+        h += Mat3::from_cols(dc * sc.x, dc * sc.y, dc * sc.z);
```
it cannot be done first as dc * sc instead component by component?

AnilKumarSingh9856 commented:
@sidd-27 @cjpurackal @edgarriba 
Thanks for the refactoring tips! I've updated the PR to use Vec3 and Mat3 directly, which really cleaned up the covariance calculation.
I also corrected the determinant-flip logic and added the complex rotation test you suggested.
I've cleaned up the code to pass the linter. It looks like the core svd3 accuracy issue is the remaining blocker for the epnp tests.

cjpurackal commented:
@AnilKumarSingh9856 Do you want to take a look into the accuracy issue of svd3? I don't think we can merge this until that is solved. For context, the 3d-2d correspondence points in the failing test are derived from opencv's epnp method.

edgarriba commented on crates/kornia-linalg/src/rigid.rs:
```diff
@@ -34,76 +27,60 @@ pub fn umeyama(src: &[Vec3], dst: &[Vec3]) -> UmeyamaResult {
+
+    // Clone to get owned Mat3 regardless of whether svd3 returned &Mat3 or Mat3.
+    // If svd3 returns a Result or other wrapper you will need to adapt this.
+    let mut u_mat = svd_result.u().clone();
```
why clone ? same bleow
